### PR TITLE
Make GRANT to non-existing role return meaningful exception

### PR DESCRIPTION
**Pull request by skrzypo987**

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            public ConnectorMetadata getMetadata(ConnectorTransactionHandle transaction)
+            {
+                return new SystemTablesMetadataMock(new StaticSystemTablesProvider(ImmutableSet.of()));
+            }
+        };
+    }
+
+    private static class SystemTablesMetadataMock
+            extends
+            SystemTablesMetadata
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:95]: Formatting is odd. Looks like someone is using new computer.

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                createTestSessionConnector(),
+                createInformationSchemaCatalogName(catalog),
+                createTestSessionConnector(),
+                createSystemTablesCatalogName(catalog),
+                createTestSessionConnector());
+    }
+
+    private static Connector createTestSessionConnector()
+    {
+        return new Connector()
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:77]: Can you use `MockConnector`?

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java (hunk 1 above)
Comment (Praveen2112) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:95]: Can we move it in line with extends

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java (hunk 2 above)
Comment (Praveen2112) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:77]: Can we use `MockConnector` ?

File: presto-main/src/main/java/io/prestosql/execution/GrantTask.java [hunk 3]
```diff
@@ -53,15 +58,34 @@ public String getName()
     @Override
     public ListenableFuture<?> execute(Grant statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
```
Comment (kokosing) [inline presto-main/src/main/java/io/prestosql/execution/GrantTask.java:59]: Why not to use this method in tests like it is used in `io.prestosql.execution.TestCallTask`?

**Comment by kokosing** [comment]
Is there any issue reported in Github that you are fixing?

**Comment by skrzypo987** [comment]
> Is there any issue reported in Github that you are fixing?


Half of https://github.com/prestosql/presto/issues/5006

**Comment by kokosing** [comment]
What about other half? Is it actually fixable?

**Comment by skrzypo987** [comment]
I don't think so. We cannot magically get user list. It is out of scope of the connector.
Or I may be missing something.

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java (hunk 1 above)
Comment (skrzypo987) [reply presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:95]: Nope, this is a human error. My formatting options are fine.

File: presto-main/src/main/java/io/prestosql/execution/GrantTask.java (hunk 3 above)
Comment (skrzypo987) [reply presto-main/src/main/java/io/prestosql/execution/GrantTask.java:59]: CallTask uses stateMachine argument for something more than getSession method. This one (and most other tested tasks) do not.

**Comment by skrzypo987** [comment]
Used MockConnector and fixed formatting.
Some tests will still fail.

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        assertThatThrownBy(() -> new GrantTask().execute(statement, transactionManager, metadata, accessControl, testSession, ImmutableList.of()))
+                .isInstanceOf(PrestoException.class)
+                .hasMessageContaining("Role 'non-existent' does not exist");
+    }
+
+    public static Catalog createBogusTestingCatalog(String catalogName)
+    {
+        Connector connector = MockConnectorFactory.builder().build()
+                .create(catalogName, null, null);
+        CatalogName catalog = new CatalogName(catalogName);
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:71]:
```
        Connector connector = MockConnectorFactory.builder()
                .build()
                .create(catalogName, null, null);
```

Let's keep a single method call per line.

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.sql.tree.PrincipalSpecification.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestGrantTask
+{
+    @Test
+    public void testGrantToNonExistingRole()
```
Comment (kokosing) [inline presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:48]: Would it be possible to have a positive test here?

**Review by kokosing — APPROVED** [review]

File: presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java (hunk 5 above)
Comment (skrzypo987) [reply presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java:48]: You asked so nicely that I can't resist.

**Comment by mosabua** [comment]
:wave: @skrzypo987 @kokosing - this PR has become inactive. If you're still interested in working on it, please let us know.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by colebow** [comment]
I'm interpreting the periodic /keep_alive as a sign that we shouldn't close this one. 😛

**Comment by bitsondatadev** [comment]
Agreed, I'll ping @kokosing

**Comment by dain** [comment]
Not sure, but I think this will need to be updated for the system roles added last year

**Comment by dain** [comment]
Ah this check was added when system roles were introducted: https://github.com/trinodb/trino/blob/eb5b77d44ab22713b249f709faced39f0b25620e/core/trino-main/src/main/java/io/trino/execution/GrantRolesTask.java#L90

**Pull request closed without merging**